### PR TITLE
[MLIR][LLVM] Translate Debug EmissionKind

### PR DESCRIPTION
**Pull request by zyx-billy**
Translate debug emission kind into LLVM (the importer already supports this).

### CODE REVIEW

File: mlir/test/Target/LLVMIR/llvmir-debug.mlir [hunk 1]
```diff
@@ -286,3 +286,24 @@ llvm.func @dbg_intrinsics_with_no_location(%arg0: i32) -> (i32) {
...
+// -----
+
+// CHECK-DAG: ![[FILE1:.*]] = !DIFile(filename: "foo1.mlir", directory: "/test/")
+// CHECK-DAG: ![[FILE2:.*]] = !DIFile(filename: "foo2.mlir", directory: "/test/")
+#di_file_1 = #llvm.di_file<"foo1.mlir" in "/test/">
+#di_file_2 = #llvm.di_file<"foo2.mlir" in "/test/">
+// CHECK-DAG: ![[SCOPE1:.*]] = distinct !DICompileUnit(language: DW_LANG_C, file: ![[FILE1]], producer: "MLIR", isOptimized: true, runtimeVersion: 0, emissionKind: LineTablesOnly)
+// CHECK-DAG: ![[SCOPE2:.*]] = distinct !DICompileUnit(language: DW_LANG_C, file: ![[FILE2]], producer: "MLIR", isOptimized: true, runtimeVersion: 0, emissionKind: DebugDirectivesOnly)
+#di_compile_unit_1 = #llvm.di_compile_unit<sourceLanguage = DW_LANG_C, file = #di_file_1, producer = "MLIR", isOptimized = true, emissionKind = LineTablesOnly>
+#di_compile_unit_2 = #llvm.di_compile_unit<sourceLanguage = DW_LANG_C, file = #di_file_2, producer = "MLIR", isOptimized = true, emissionKind = DebugDirectivesOnly>
```
Comment (gysit) [inline mlir/test/Target/LLVMIR/llvmir-debug.mlir:299]:
```suggestion
// CHECK-DAG: ![[FILE1:.*]] = !DIFile(filename: "foo1.mlir", directory: "/test/")
#di_file_1 = #llvm.di_file<"foo1.mlir" in "/test/">
// CHECK-DAG: ![[FILE2:.*]] = !DIFile(filename: "foo2.mlir", directory: "/test/")
#di_file_2 = #llvm.di_file<"foo2.mlir" in "/test/">
// CHECK-DAG: ![[SCOPE1:.*]] = distinct !DICompileUnit(language: DW_LANG_C, file: ![[FILE1]], producer: "MLIR", isOptimized: true, runtimeVersion: 0, emissionKind: LineTablesOnly)
#di_compile_unit_1 = #llvm.di_compile_unit<sourceLanguage = DW_LANG_C, file = #di_file_1, producer = "MLIR", isOptimized = true, emissionKind = LineTablesOnly>
// CHECK-DAG: ![[SCOPE2:.*]] = distinct !DICompileUnit(language: DW_LANG_C, file: ![[FILE2]], producer: "MLIR", isOptimized: true, runtimeVersion: 0, emissionKind: DebugDirectivesOnly)
#di_compile_unit_2 = #llvm.di_compile_unit<sourceLanguage = DW_LANG_C, file = #di_file_2, producer = "MLIR", isOptimized = true, emissionKind = DebugDirectivesOnly>
```
ultra nit: I would try to put the check line right above the line you are matching.

File: mlir/lib/Target/LLVMIR/DebugTranslation.cpp [hunk 2]
```diff
@@ -102,7 +102,9 @@ llvm::DICompileUnit *DebugTranslation::translateImpl(DICompileUnitAttr attr) {
       attr.getSourceLanguage(), translate(attr.getFile()),
       attr.getProducer() ? attr.getProducer().getValue() : "",
       attr.getIsOptimized(),
-      /*Flags=*/"", /*RV=*/0);
+      /*Flags=*/"", /*RV=*/0, /*SplitName=*/{},
+      static_cast<llvm::DICompileUnit::DebugEmissionKind>(
```
Comment (gysit) [inline mlir/lib/Target/LLVMIR/DebugTranslation.cpp:106]: I wonder why we did not choose an LLVM enum here... That way we would have a guarantee that the two enum values match. However, this is orthogonal to your revision.

**Review by gysit — APPROVED** [review]
Thanks for fixing! 

LGTM

File: mlir/lib/Target/LLVMIR/DebugTranslation.cpp (hunk 2 above)
Comment (zyx-billy) [reply mlir/lib/Target/LLVMIR/DebugTranslation.cpp:106]: Yeah there are other enums like `DISubprogramFlags` that are also just redefined and rely on static_cast. Might be a good idea to reuse llvm enums if possible, or set up explicit mappings?

Comment (gysit) [reply mlir/lib/Target/LLVMIR/DebugTranslation.cpp:106]: Yeah I guess I would prefer if we use LLVM enums since there are no large switch statements needed (I assume that is what you refer to by explicit mappings)? But this is unrelated to your PR. I just realized that things are a bit inconsistent in that regard.

Comment (zyx-billy) [reply mlir/lib/Target/LLVMIR/DebugTranslation.cpp:106]: Yeah that's what I meant. That makes sense. We can leave it at this then for this PR. Can you help me click merge once it's ready? Thanks! 🙏

Comment (gysit) [reply mlir/lib/Target/LLVMIR/DebugTranslation.cpp:106]: sure!
